lib/spack/spack/cmd/uninstall.py

```diff
@@ -225,15 +225,25 @@ def do_uninstall(env, specs, force):
 
     # A package is ready to be uninstalled when nothing else references it,
     # unless we are requested to force uninstall it.
-    is_ready = lambda x: not spack.store.db.query_by_spec_hash(x)[1].ref_count
-    if force:
-        is_ready = lambda x: True
+    def is_ready(dag_hash):
+        if force:
+            return True
+
+        _, record = spack.store.db.query_by_spec_hash(dag_hash)
+        if not record.ref_count:
+            return True
+
+        # if only remaining depenents are build dependencies, we can uninstall
```

Comment:
Maybe:
```suggestion
        # If this spec is only used as a build dependency, we can uninstall
```
is a bit clearer?